tmt/result.py

```diff
@@ -310,9 +318,82 @@ def interpret_result(self, interpret: ResultInterpret) -> 'Result':
         :returns: :py:class:`Result` instance containing the updated result.
         """
 
-        if interpret in (ResultInterpret.RESPECT, ResultInterpret.CUSTOM):
+        if interpret == ResultInterpret.CUSTOM:
             return self
 
+        # Check for failed checks
+        checks_failed = any(check.result == ResultOutcome.FAIL for check in self.check)
+
+        if interpret == ResultInterpret.RESPECT:
+            if checks_failed:
+                self.result = ResultOutcome.FAIL
+                if self.note:
+                    self.note += ', check failed'
+                else:
+                    self.note = 'check failed'
+            return self
+
+        # Handle XFAIL for the overall test result
+        if interpret == ResultInterpret.XFAIL:
+            if self.result == ResultOutcome.PASS and checks_failed:
+                self.result = ResultOutcome.FAIL
```

Comment:
@psss I got a little entangled in the xfail logic and the `# Handle individual check results` part should probably be done before this.   

I wonder if it will be digestible enough for the users, I mean:
 xfail test fails, check with default 'respect' fails = FAIL
 i.e. users would have to add xfail also to check, if expected. Could be breaking change for some?

Reply:
Thanks for doing this. I would also like to petition for the following scenario.
if the before test portion is the only part that fails for the check, the test should be `self.result = ResultOutcome.WARN`.